bravado/http_future.py

```diff
@@ -58,32 +83,18 @@ def result(self, timeout=None):
 
 
 def reraise_errors(func):
+
     @wraps(func)
     def wrapper(self, *args, **kwargs):
-        timeout_errors = tuple(getattr(self.future, 'timeout_errors', None) or ())
-
-        # Make sure that timeout error type for a specific future adapter is generated only once
-        if timeout_errors and getattr(self.future, '__timeout_error_type', None) is None:
-            setattr(
-                self.future, '__timeout_error_type',
-                type(
-                    '{}Timeout'.format(self.future.__class__.__name__),
-                    tuple(list(timeout_errors) + [BravadoTimeoutError]),
-                    dict(),
-                ),
-            )
+        timeout_errors = tuple(self.future.timeout_errors or ())
+        connection_errors = tuple(self.future.connection_errors or ())
```

Comment:
Could we make this code backwards compatible? I think something like

    connection_errors = getattr(self.future, 'connection_errors', ())

I don't think it's a problem to use whatever the value of that attribute is if it is defined.

Reply:
Sure ... keeping backward compatibility is something to aim for :)
I'm not against using `getattr` but I would understand how the change could provide backward incompatible changes (so next time I'll think to it too)

Reply:
We talked about it and well the change is backward compatible as `self.future` in an instance of `bravado.http_future. FutureAdapter` which defines those attributes.
So even if http clients are not defining them the attribute will be available with an empty tuple as default value